libs/react-spa/shared/src/lib/loaders/loaders.ts

```diff
@@ -0,0 +1,109 @@
+import { z } from 'zod'
+
+/**
+ * Creates one level deep object from a URLSearchParams
+ * Each key is split by dot notation and the value is the value of the key
+ *
+ * @example
+ * createSearchParamObj(new URLSearchParams('age.from=1&age.to=2&flightLeg.from=KEF&flightLeg.to=AER&period.from=2021-01-01&period.to=2021-01-02&state=active&state=inactive&isExplicit=true&airline=W6'))
+ * {
+ *   'age.from': 1,
+ *   'age.to': 2,
+ *   'flightLeg.from': 'KEF',
+ *   'flightLeg.to': 'AER',
+ *   'period.from': '2021-01-01',
+ *   'period.to': '2021-01-02',
+ *   state: ['active', 'inactive'],
+ *   isExplicit: 'true',
+ *   airline: 'W6',
+ * }
+ */
+export const createSearchParamObj = (searchParams: URLSearchParams) => {
```

Comment:
~Might want to keep some of these helpers private (not exported from `@island.is/react-spa/shared`) unless there's a use case for it.

Reply:
Its used in tests.

Reply:
Yeah, but the overall package can handpick exports:

```tsx
export { validateSearchParams, validateFormData } from './lib/validate.ts'
```